### PR TITLE
python3

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: bootstrap [hunk 1]
```diff
@@ -25,15 +25,15 @@ if [ -f /etc/redhat-release ]; then
...
 # avoid pip bugs
 ./virtualenv/bin/pip install --upgrade pip
 
 # work-around change in pip 1.5
-./virtualenv/bin/pip install six
-./virtualenv/bin/pip install -I nose
-./virtualenv/bin/pip install setuptools==32.3.1
+#./virtualenv/bin/pip install six
+#./virtualenv/bin/pip install -I nose
+#./virtualenv/bin/pip install setuptools==32.3.1
```
Comment (alfredodeza) [inline bootstrap]: are you sure that setuptools is not needed anymore?

File: bootstrap [hunk 2]
```diff
@@ -14,7 +14,7 @@ if [ -f /etc/debian_version ]; then
     fi
 fi
 if [ -f /etc/redhat-release ]; then
-    for package in python-pip python-virtualenv python-devel python-six libevent-devel libxml2-devel libxslt-devel zlib-devel; do
+    for package in python3-pip python3-virtualenv python3-devel python3-six libevent-devel libxml2-devel libxslt-devel zlib-devel; do
```
Comment (alfredodeza) [inline bootstrap]: python3-six wouldn't be needed anymore, it is mostly for backwards compatibility between py2/py3

File: bootstrap [hunk 3]
```diff
@@ -2,7 +2,7 @@
 set -e
 
 if [ -f /etc/debian_version ]; then
-    for package in python-pip python-virtualenv python-dev python-six libevent-dev libxml2-dev libxslt-dev zlib1g-dev; do
+    for package in python3-pip python3-virtualenv python3-dev python3-six libevent-dev libxml2-dev libxslt-dev zlib1g-dev; do
```
Comment (alfredodeza) [inline bootstrap]: I'd remove python3-six

File: bootstrap [hunk 4]
```diff
@@ -25,15 +25,15 @@ if [ -f /etc/redhat-release ]; then
...
 
 # avoid pip bugs
 ./virtualenv/bin/pip install --upgrade pip
 
 # work-around change in pip 1.5
-./virtualenv/bin/pip install six
-./virtualenv/bin/pip install -I nose
-./virtualenv/bin/pip install setuptools==32.3.1
+#./virtualenv/bin/pip install six
+#./virtualenv/bin/pip install -I nose
```
Comment (alfredodeza) [inline bootstrap]: if nose is not installed how are the imports going to work?

**Comment by liewegas** [comment]
updated

**Review by alfredodeza — APPROVED** [review]

**Comment by liewegas** [comment]
@yehudasa I'm running up against changes to configparser in py2 vs py3.. afaics it doesn't do the arbitrarily nested levels that ragweed is doing.  TBH I'm a bit confused by what ragweed.py is doing here: https://github.com/ceph/ceph/blob/master/qa/tasks/ragweed.py#L125 .. it looks like it's nested several levels?  I wonder if it would be simpler to just change this to read and write a blob of JSON instead of an ini file for the master branch?

FWIW i'm testing this by force-pushing to wip-sage3-testing-2019-12-17-1443 in ragweed.git and scheduling a run like so:
```
teuthology-suite -s rgw -c wip-sage3-testing-2019-12-17-0629 --suite-repo https://github.com/liewegas/ceph --suite-branch wip-sage3-testing-2019-12-17-1443 -p 30 --subset 1/999 -m smithi  --filter ragweed --limit 2
```

**Comment by yehudasa** [comment]
@liewegas it's a copy-paste of what the s3tests task is doing here:
https://github.com/ceph/ceph/blob/master/qa/tasks/s3tests.py#L98
The task allows configuring different users for different clients. I'm not sure it's a functionality we want to lose. This is the task yaml config, not the ragweed configuration itself, I'm not sure what you mean by the blob of json suggestion.

**Comment by yehudasa** [comment]
@liewegas looking at the recent failures here:
http://pulpito.ceph.com/sage-2019-12-17_22:02:38-rgw-wip-sage3-testing-2019-12-17-0629-distro-basic-smithi/

It's a problem with the use of the Bunch package. Had similar issues with s3tests, there we could just remove it. Ragweed uses it more heavily at its core. We might be able to replace it with some other package (Munch maybe).

**Pull request merged**